lis/dataassim/obs/NASA_SMAPsm/read_NASASMAPsm.F90

```diff
@@ -863,22 +879,38 @@ subroutine read_SMAPL2sm_data(n, k,fname, smobs_inp, time)
         sm_data(ease_col(t) + &
              (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = sm_field(t)
 
-        if(vwc_field(t).gt.5) then !YK Aply QC : if VWC > 5 kg/m2
-           sm_data(ease_col(t) + &
-                (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = LIS_rc%udef
-        else
-           if(sm_data(ease_col(t) + &
-                   (ease_row(t)-1)*NASASMAPsm_struc(n)%nc).ne.-9999.0) then
-              if(ibits(sm_qa(t),0,1).eq.0) then
-                 sm_data_b(ease_col(t) + &
-                    (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = .true.
-              else
-                 sm_data(ease_col(t) + &
-                      (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = LIS_rc%udef
-              endif
-           endif
-        endif
-     endif
+        ! ZH: commented out VWC if-statement since it is redundant:
+        ! ibits(sm_qa(t),0,1) will equal 1 if VWC > 5 kg/m2
+        !
+        !if(vwc_field(t).gt.5) then !YK Aply QC : if VWC > 5 kg/m2
+        !   sm_data(ease_col(t) + &
+        !        (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = LIS_rc%udef
+        !else
+        !
+         if(sm_data(ease_col(t) + &
+                  (ease_row(t)-1)*NASASMAPsm_struc(n)%nc).ne.-9999.0) then
+            if(ibits(sm_qa(t),0,1).eq.0) then
+            ! retrieval has the recommended quality 
+            ! (not the case when VWC > 5 kg/m2, and/or other reasons)
+               sm_data_b(ease_col(t) + &
+                  (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = .true.
+            ! ZH: check if it is only the dense vegetation (bit 10) causing the retrieval
+            ! not to have the recommended quality, still assimilate in this case
+            elseif(NASASMAPsm_struc(n)%vegetation_flag.eq.0 .and. ibits(sm_surface(t),0,1).eq.0 .and. &
+               ibits(sm_surface(t),1,1).eq.0 .and. ibits(sm_surface(t),2,1).eq.0 .and. &
+               ibits(sm_surface(t),3,1).eq.0 .and. ibits(sm_surface(t),4,1).eq.0 .and. &
+               ibits(sm_surface(t),5,1).eq.0 .and. ibits(sm_surface(t),6,1).eq.0 .and. &
+               ibits(sm_surface(t),7,1).eq.0 .and. ibits(sm_surface(t),8,1).eq.0 .and. &
+               ibits(sm_surface(t),9,1).eq.0) then
```

Comment:
Surface flag has 15 bits; 11-15 not used; and 10 is vegetation - which can be anything. So here we say that if a retrieval is attempted, the retrieval can be used, but we lost info on whether the retrieval was successful. I think we need to add "if ibits(sm_qa(t),0,3).eq.0 " as well? Or, perhaps only "if ibits(sm_qa(t),0,3).eq.0 " is enough? I think (but might be wrong) that the current elseif statement is essentially equivalent to "if ibits(sm_qa(t),0,2).eq.0 ", no?

Reply:
What do you mean with "10 is vegetation - which can be anything"? In the user guide it is specified that if VWC < 5, then bit 10 equals 0, and 1 otherwise, so that's quite specific?
![image](https://github.com/KUL-RSDA/LISF/assets/11814609/407a254f-a3e2-4360-883c-9727e8ca1089)
![image](https://github.com/KUL-RSDA/LISF/assets/11814609/b925016d-2aa7-462d-92e5-f3dbe5235d31)

I agree that I didn't make a distinction between attempted and successful retrievals so far. I will add this to the implementation. But I'm not sure about your suggestion. Don't you mean we should add the statement `if ibits(sm_qa(t),2,1).eq.0` and that our current elseif statement is equivalent to `if ibits(sm_qa(t),1,1).eq.0` (see table)?

![image](https://github.com/KUL-RSDA/LISF/assets/11814609/dd3f3e2d-8013-4b61-aadf-f2fa487f0906)

Reply:
Indeed, I would suggest adding ibits(sm_qa(t),2,1).eq.0 as an extra condition when the vegetation_flag is off (and when there is no check on the VWC).